adagrams/game.py

```diff
@@ -1,11 +1,119 @@
+"""Adagrams: It's really scrabble"""
+
+import random
+
+LETTER_POOL = {
+    'A': 9,
+    'B': 2,
+    'C': 2,
+    'D': 4,
+    'E': 12,
+    'F': 2,
+    'G': 3,
+    'H': 2,
+    'I': 9,
+    'J': 1,
+    'K': 1,
+    'L': 4,
+    'M': 2,
+    'N': 6,
+    'O': 8,
+    'P': 2,
+    'Q': 1,
+    'R': 6,
+    'S': 4,
+    'T': 6,
+    'U': 4,
+    'V': 2,
+    'W': 2,
+    'X': 1,
+    'Y': 2,
+    'Z': 1
+}
+
+letter_value = {
+    1: ['A', 'E', 'I', 'O', 'U', 'L', 'N', 'R', 'S', 'T'],
+    2: ['D', 'G'],
+    3: ['B', 'C', 'M', 'P'],
+    4: ['F', 'H', 'V', 'W', 'Y'],
+    5: ['K'],
+    8: ['J', 'X'],
+    10: ['Q', 'Z']
+}
+
+
 def draw_letters():
-    pass
+    """Output a list of ten strings, where each string is a letter of 
+    len(1) as defined in LETTER_POOL."""
+
+    letter_pool_copy = LETTER_POOL.copy()
+    letters = []
+
+    while len(letters) < 10:
+        letter_key = random.choice(list(letter_pool_copy.keys()))
+        if letter_pool_copy[letter_key] != 0:
+            letter_pool_copy[letter_key] -= 1
+            letters.append(letter_key)
+
+    return letters
+
 
 def uses_available_letters(word, letter_bank):
```

Comment:
This works fine! We've only just talked about Big O and time/space complexity, but one thing to notice here is we have a 3 nested for loops, even if we haven't explicitly stated `for___ in ___` three times.

The `remove` method loops through `letter_bank_case` and `if letter in letter_bank_case` loops under the hood. Nothing wrong with 2 nested for loops. At times they are the easier solution. But! we have 3 here, and a dictionary of letters counted from the `letter_bank_case` would help us here.

How might we create a "frequency map" (a dictionary of items counted) with a constant lookup (O(1)) of letters found in `letter_bank_case`, then use it to check if all those letters are in the `word`?